addons/mail/models/mail_link_preview.py

```diff
@@ -142,3 +147,15 @@ def _gc_mail_link_preview(self):
             ('message_id', '=', False),
             ('create_date', '<', fields.Datetime.now() - timedelta(days=lifetime)),
         ], order='create_date ASC', limit=1000).unlink()
+
+    @api.model
+    def _init_ignore_list(self):
+        try:
+            base_url = urlparse(self.get_base_url())
+            base_ignore_pattern = f".*{base_url.hostname}"
+            if base_url.port:
+                base_ignore_pattern = f"{base_ignore_pattern}:{base_url.port}"
+            base_ignore_pattern = f"{base_ignore_pattern}/(web|odoo)" # ignore backend url by default
+            self.env["ir.config_parameter"].set_param("mail.mail_link_preview_ignore_list", base_ignore_pattern)
+        except:
+            pass
```

Comment:
What line can trigger which exception? We usually never blind except/ignore as it will catch anything, including programming error and other unexpected things.

The only thing we really want to handle is invalid format in base url.

Reply:
Alright, I updated the try/except